### ISSUE TITLE
Case of unstable initial condition finding with BrownFullBasicInit 

### ISSUE THREAD

**Issue by AnHeuermann** [SciML/OrdinaryDiffEq.jl#1498]
This issue is from a [discussion on discourse.julialang.org](https://discourse.julialang.org/t/modelingtoolkit-index-reduction-and-linear-algebraic-loops/69312/3) about ModelingToolkit.jl and that I used wrong start values for my ODE problem.

**In short**

I'm using the wrong start values `u0` for `x` and `y` and during simulation I'll get an error:
```julia
using ModelingToolkit
using OrdinaryDiffEq

@parameters t
@variables x(t) y(t) a(t) b(t)
der = Differential(t)

eqs = [der(x) ~ y + a,
       0 ~ 3*x - y,
       der(y) ~ a*x,
       0 ~ a/5 - b]

@named sys = ODESystem(eqs,t)

sys = dae_index_lowering(sys)
sys = structural_simplify(sys)

u0 = [y => 1.0,
      x => 3.0,
      a => 0.0]
p = []

tspan = (0.0,1.0)
prob = ODEProblem(sys, u0, tspan, p)
sol = solve(prob, Rodas4())
```

```
 Warning: Instability detected. Aborting
└ @ SciMLBase C:\Users\...\.julia\packages\SciMLBase\n3U0M\src\integrator_interface.jl:351
```

At that time I could not find the source of the error.

--------------------

My suggestion would be to improve the error message so that a user gets an error like
```
Error: The initialization problem is inconsistent due to the following equation: 0 != 8 = 3*x - y
Aborting
```
(Error message in parts taken from OpenModelica).

I would argue for an error over a warning because then the user needs to fix the model and can't simply ignore the problem only the see it pop up in the next tool again.

**Comment by ChrisRackauckas** [comment]
That's not the issue. As mentioned on Discourse, this solver can automatically find initial conditions. The issue is that if you check the return:

```julia
(solve(prob, Rodas4()))[1] = [3.0, 1.0, -Inf]
```

for some reason the initial condition finder failed and return a -Inf. So it's an error there.

Did you try the Shampine initialization?

**Comment by ChrisRackauckas** [comment]
We should add a warn and exit if the initialization fails though.

**Comment by AnHeuermann** [comment]
The wrong solution is probably because of `sys = dae_index_lowering(sys)` and not an issue of OrdinaryDiffEq.jl but SciML/ModelingToolkit.jl.
It is simply that an important equation `y=3*x` from the original system is missing in the reduced system.
![image](https://user-images.githubusercontent.com/38031952/136378473-bc276582-c391-4fa2-99d4-98a756634321.png)

The dummy-derivate method would be one way to solve this. 
I can open a ticket about this on SciML/ModelingToolkit.jl.

**Comment by ChrisRackauckas** [comment]
Yes, that looks like it could be what's going on. Nice snooping.
